apps/frontend/src/pages/people/[username].tsx

```diff
@@ -206,34 +206,57 @@ const UserProfilePage: React.FC = () => {
     tensions: string[];
     contacts: string[];
   } | null>(null);
+  const [isUnregistered, setIsUnregistered] = useState(false);
   const { socket } = useSocket();
 
   useEffect(() => {
     const fetchConnectionAndTapInfo = async () => {
       if (typeof username === "string") {
         const user = await storage.getUser();
         const session = await storage.getSession();
-        if (!user || !session || !user.connections[username]) {
+        const unregisteredUser = await storage.getUnregisteredUser();
+        if ((!user || !session || !user.connections[username]) && !unregisteredUser) {
           console.error("User not found");
           toast.error("User not found");
           router.push("/people");
           return;
         }
 
-        setUser(user);
-        setSession(session);
-        setConnection(user.connections[username]);
-        const savedTapInfo = await storage.loadSavedTapInfo();
-        // Delete saved tap info after fetching
-        await storage.deleteSavedTapInfo();
-        // Show tap modal if tap info is for current connection
-        if (
-          savedTapInfo &&
-          savedTapInfo.tapResponse.userTap?.ownerUsername === username
-        ) {
-          logClientEvent("user-profile-tap-chip-modal-shown", {});
-          setTapInfo(savedTapInfo);
-          setShowCommentModal(true);
+        if (unregisteredUser) {
+          setIsUnregistered(true);
+        }
+
+        if (user && session) {
+          setUser(user);
+          setSession(session);
+          setConnection(user.connections[username]);
+          const savedTapInfo = await storage.loadSavedTapInfo();
+          // Delete saved tap info after fetching
+          await storage.deleteSavedTapInfo();
+          // Show tap modal if tap info is for current connection
+          if (
+            savedTapInfo &&
+            savedTapInfo.tapResponse.userTap?.ownerUsername === username
+          ) {
+            logClientEvent("user-profile-tap-chip-modal-shown", {});
+            setTapInfo(savedTapInfo);
+            setShowCommentModal(true);
+          }
+        } else if (unregisteredUser) {
+          // In unregistered user case, cannot do PSIs which require user object
+          setConnection(unregisteredUser.connections[username]);
+          const savedTapInfo = await storage.loadSavedTapInfo();
```

Comment:
I see, the earlier saveTpa it's used to transfer over info to this page, makes sense